include/rta/fullsupernodelist.h

```diff
@@ -138,6 +152,8 @@ class FullSupernodeList
     mutable boost::shared_mutex m_access;
     std::unique_ptr<utils::ThreadPool> m_tp;
     std::atomic_size_t m_refresh_counter;
+    uint64_t m_blockchain_based_list_block_number;
+    SupernodeArray m_blockchain_based_list; //TODO: lifetime of supernodes, should this be an array of raw pointers?
```

Comment:
To partially address the TODO: No: Raw pointers are almost never the right answer in modern C++ code.  If you need a weak ownership, that's what `std::weak_ptr` is for.  (Or `boost::weak_ptr`, but it's rather strange to me that SupernodePtr is defined as a `boost::shared_ptr` when the only practical purpose for that nowadays is to have a `std::shared_ptr`-like class for pre-C++11 codebases—which doesn't apply here).

Reply:
You are absolutely right. The meaning was related to lifetime control of Supernode object. I'm still thinking about management of this collection.